File: 4. Reproducible Code/mod_prog_py_r/course_content/example_code_python/case_study/initial/input_output.py
import pandas as pd
import os


def load_formatted_frame(path_to_data):
    """Read csv and reformat column names"""
    formatted_path = os.path.join(path_to_data)
    # Load the csv from given path
    dataframe = pd.read_csv(formatted_path)
    
    # Clean the column names, following naming conventions similar to PEP8
    dataframe.columns = dataframe.columns.str.lower()
    dataframe.columns = dataframe.columns.str.replace(" ", "_")
    
    return dataframe
    

def write_output(dataframe, output_filepath):
    """Function to write output statistic in formatted manner"""

    dataframe.to_csv(output_filepath, index=False, sep=",")
    

File: 4. Reproducible Code/mod_prog_py_r/course_content/example_code_R/case_study/initial/input_output.r
library(readr)

#' Read population data and reformat column names
load_formatted_frame <- function(path_to_data) {
    # Load the population density data 2019
    formatted_path <- file.path(path_to_data)
    dataframe <- readr::read_csv(formatted_path)
    
    # Clean the column names, following snake_case convention
    colnames(dataframe) <- tolower(colnames(dataframe))
    colnames(dataframe) <- stringr::str_replace_all(colnames(dataframe), pattern = " ", replacement = "_")
    
    return(dataframe)
}
    

#' write output statistic in formatted manner
write_output <- function(dataframe, output_filepath) {
    
    readr::write_csv(x = dataframe, path = output_filepath)
    
}
